src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2449,207 +2332,242 @@ private bool ConsumeSingleLineCommentMultiSegment(ReadOnlySpan<byte> localBuffer
-                            }
-                            HasValueSequence = true;
-                            _totalConsumed += localBuffer.Length + leftOver;
+                    expectSlash = false;
+                }
 
-                            if (lastLineFeedIndex == -1)
-                            {
-                                _bytePositionInLine += localBuffer.Length + leftOver;
-                            }
-                            else
-                            {
-                                _bytePositionInLine += i - lastLineFeedIndex - 1;
-                            }
-                            lastLineFeedIndex = -1;
-                            leftOver = 0;
-                            previousConsumed = 0;
-                            localBuffer = _buffer;
-                            i = 0;
-                            if (localBuffer[i] == JsonConstants.Slash)
-                            {
-                                goto Done;
-                            }
-                            break;
-                        }
-                    }
-                    else if (nextByte == JsonConstants.LineFeed)
-                    {
-                        lastLineFeedIndex = i;
-                        _bytePositionInLine = 0;
-                        _lineNumber++;
-                        lastAsterisk = false;
-                    }
-                    else
+                if (ignoreNextLfForLineTracking)
+                {
+                    if (localBuffer[0] == JsonConstants.LineFeed)
                     {
-                        lastAsterisk = false;
+                        _consumed++;
+                        localBuffer = localBuffer.Slice(1);
                     }
+
+                    ignoreNextLfForLineTracking = false;
                 }
-                if (i == localBuffer.Length)
+
+                int idx = localBuffer.IndexOfAny(JsonConstants.Asterisk, JsonConstants.LineFeed, JsonConstants.CarriageReturn);
+
+                if (idx != -1)
                 {
-                    if (!GetNextSpan())
+                    int nextIdx = idx + 1;
+                    byte marker = localBuffer[idx];
+                    localBuffer = localBuffer.Slice(nextIdx);
+
+                    _consumed += nextIdx;
+
+                    switch (marker)
                     {
-                        _totalConsumed = prevTotalConsumed;
-                        return false;
+                        case JsonConstants.Asterisk:
+                            expectSlash = true;
+                            _bytePositionInLine += nextIdx;
+                            break;
+                        case JsonConstants.LineFeed:
+                            _bytePositionInLine = 0;
+                            _lineNumber++;
+                            break;
+                        default: // JsonConstants.CarriageReturn:
+                            _bytePositionInLine = 0;
```

Comment:
nit: add `Debug.Assert(marker == JsonConstants.CarriageReturn)` and remove the comment.